docs/tutorial.md

```diff
@@ -0,0 +1,369 @@
+
+    public List<String> getNodes() {
+        if (nodes == null) {
+            nodes = new ArrayList<>();
+        }
+        return nodes;
+    }
+
+    public void setNodes(List<String> nodes) {
+        this.nodes = nodes;
+    }
+}
+```
+
+**Note** The Node field is just to illustrate an example of a Status field. In real cases, it would be recommended to use [Conditions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties).
+
+#### `Memcached`
+
+`Memcached` file scaffolded via `create api` command. It extends the properties of `MemcachedSpec` and `MemcachedStatus`.
+
+```
+// Memcached is the Schema for the memcacheds API
+
+@Version("v1alpha1")
+@Group("cache.example.com")
+public class Memcached extends CustomResource<MemcachedSpec, MemcachedStatus>
+    implements Namespaced {}
+```
+
+### Apply CR and CRD's using below command
+
+#### CRD - Custom Resource
+
+Create a file with the name `crd.yaml`. CRD enables users to add their own/custom objects to the Kubernetes cluster.
+
+This file will contain the below content. 
+
+```
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.4.1
+  name: memcacheds.cache.example.com
+spec:
+  group: cache.example.com
+  names:
+    kind: Memcached
+    listKind: MemcachedList
+    plural: memcacheds
+    singular: memcached
+  scope: Namespaced
+  versions:
+  - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        ...
+    served: true
+    storage: true
+    subresources:
+      status: {}
+status:
+  acceptedNames:
+    kind: ""
+    plural: ""
+  conditions: []
+  storedVersions: []
+```
+
+Create the CRD:
+
+`kubectl apply -f cr.yaml`
+
+### Create Memcached Custom Resource - memcached-sample.yaml
+
+Update the sample Memcached CR manifest at k8s/samples/memcached-sample.yaml and define the spec as the following
+
+This file will contain the below content. 
+
+```
+apiVersion: cache.example.com/v1alpha1
+kind: Memcached
+metadata:
+  name: memcached-sample
+spec:
+  # Add fields here
+  size: 1
+```
+
+Create the CR:
+
+`kubectl apply -f memcached-sample.yaml`
+
+## Implement the Controller
+
+**Note**: The next two subsections explain how the two methods `createOrUpdateResource` and `deleteResource`. These two methods will be called whenever there is some update/create/delete event occurs in the cluster. 
+
+These methods are already scaffolded as part of the `create api` command.
+
+### createOrUpdateResource
+
+This method initially checks for the deployment by using the below code.
+
+```
+        Deployment deployment =
+                client
+                        .apps()
+                        .deployments()
+                        .inNamespace(resource.getMetadata().getNamespace())
+                        .withName(resource.getMetadata().getName())
+                        .get();
+```
```

Comment:
This is an example of you telling me the reader what the code does but you didn't tell me where I should add this and why I even care to do this. Basically the way I would approach the "implement the controller" section is as follows:

* Summarize what you plan on having the user do
* Show them how to do

For example, you would tell the user that we plan to implement the controller. In this memcached example, we will need to watch the deployment so we can react to size changes. We will accomplish this in the steps below.

Then proceed to tell them what to add. 

First, let's get the `Deployment`. In the MemachedController.java, find the `createOrUpdateResource` method and add the following code below the ` // TODO: fill in logic` comment. It should look like:

```
    @Override
    public UpdateControl<Memcached> createOrUpdateResource(
        Memcached resource, Context<Memcached> context) {
        // TODO: fill in logic
        Deployment deployment =
                client
                        .apps()
                        .deployments()
                        .inNamespace(resource.getMetadata().getNamespace())
                        .withName(resource.getMetadata().getName())
                        .get();

        return UpdateControl.noUpdate();
    }  
```

For each step you outlined below do the same thing.

At the very end of this section you should have the fully populated `createOrUpdateResource` method shown at the end so that the user can refer to it in case they made any mistakes along the way.

Reply:
Added all of the above suggestions in the Controller Logic.